crates/api_crud/src/site/create.rs

```diff
@@ -41,31 +40,30 @@ impl PerformCrud for CreateSite {
   #[tracing::instrument(skip(context))]
   async fn perform(&self, context: &Data<LemmyContext>) -> Result<SiteResponse, LemmyError> {
     let data: &CreateSite = self;
-
+    let local_user_view = local_user_view_from_jwt(&data.auth, context).await?;
     let local_site = LocalSite::read(context.pool()).await?;
 
+    // BEGIN VALIDATION
+    // Make sure user is an admin; other types of users should not update site data...
+    is_admin(&local_user_view)?;
+
+    // Make sure the site hasn't already been set up...
     if local_site.site_setup {
       return Err(LemmyError::from_message("site_already_exists"));
     };
 
-    let local_user_view = local_user_view_from_jwt(&data.auth, context).await?;
-
-    let sidebar = diesel_option_overwrite(&data.sidebar);
-    let description = diesel_option_overwrite(&data.description);
-    let icon = diesel_option_overwrite_to_url(&data.icon)?;
-    let banner = diesel_option_overwrite_to_url(&data.banner)?;
+    // Check that the slur regex compiles, and returns the regex if valid...
+    let slur_regex = build_and_check_regex(&local_site.slur_filter_regex.as_deref())?;
 
-    let slur_regex = local_site_to_slur_regex(&local_site);
+    site_name_length_check(&data.name)?;
     check_slurs(&data.name, &slur_regex)?;
-    check_slurs_opt(&data.description, &slur_regex)?;
-
-    // Make sure user is an admin
-    is_admin(&local_user_view)?;
 
-    if let Some(Some(desc)) = &description {
+    if let Some(desc) = &data.description {
```

Comment:
One thing that scares me about this change, is that the diesel_option_overwrite function has a special case: it considers sending empty string as a `set the DB item to null`. In which case, this check shouldn't be run.

By removing that, it now runs these checks on empty strings, which might fail.

Reply:
You can see this working as (I believe) expected here: https://github.com/LemmyNet/lemmy/pull/3146/files#diff-d9124d8135ec751589f580dd3a84eba50f3ddb2bd27bd2cfb7a436f6af9e9c54R276